### ISSUE TITLE
Terminal Doesn't Follow Nemo

### ISSUE THREAD

**Issue by kendoori** [linuxmint/nemo-extensions#134]
Am running 17.2 Cinnamon and use the Nemo's ability to open a terminal pane (F4). At some point, I seem to have lost the functionality of Terminal opening in the current folder. Right now, regardless of where Nemo is, it opens to the top of my 'home" directory.

I've tried resetting dconf settings to default to no avail.

![selection_050](https://cloud.githubusercontent.com/assets/2447420/10060902/8bb95310-6222-11e5-96a8-f9cbf286aa5f.png)

**Comment by benwiley4000** [comment]
I can second this.

**Comment by GLolol** [comment]
Duplicate of #113.

**Comment by marco44** [comment]
Hi,

I'm still having this (or a similar, I don't know) issue.

Arch Linux, Nemo 3.2.0, Nemo-terminal 3.2.0.

Is there information I could provide ?

**Comment by Jarshvor** [comment]
@marco44 Did you get a fix for this?
not sure if this is fixed in the mentioned #113, but I'm still having this issue:
Arch Linux too, nemo 3.2.2-1 nemo-terminal 3.2.0-1

**Comment by marco44** [comment]
No, I didn't :)

**Comment by dg1727** [comment]
I am having the problem "Terminal doesn't follow Nemo."  (Whether the cause is the same or not, "terminal doesn't start in the right folder" sounds different than "the open terminal doesn't follow Nemo's folder").  The `nemo_terminal.py` from UberPanda in #113 doesn't fix the problem for me, including after I disable & re-enable the plugin in Edit > Plugins (Extensions section).  

LMDE 2 "Betsy" Cinnamon 64-bit 
`uname -rv` says `3.16.0-4-amd64 #1 SMP Debian 3.16.39-1 (2016-12-30)` 
Nemo 3.2.2, `nemo-terminal` APT package version 3.2.1+betsy

I added this comment because none of the existing comments were close enough to my situation (e.g., OS version) that I wanted to give them a thumbs-up.

**Comment by wavic** [comment]
Same here. Even F4 doesn't work. The terminal does not follow folder changes. And vice versa.

**Comment by v1993** [comment]
Same issue on XUbuntu 18.04 with cinnamon.

**Comment by claudiux** [comment]
Same issue on Archlinux 27 with Cinnamon 3.8.3.

**Comment by claudiux** [comment]
... and on LinuxMint 19 BETA with Cinnamon 3.8.3.

**Comment by johanvandegriff** [comment]
I am running Manjaro 17.1.10 with nemo 3.8.3 and I was able to fix the issue. 

I ran nemo in the terminal:
```bash
killall nemo
nemo
```
and tried to change directories to see what the error was:
```bash
Traceback (most recent call last):
  File "/usr/share/nemo-python/extensions/nemo_terminal.py", line 463, in _on_crowbar_parent_set
    nterm.change_directory(self._uri)
  File "/usr/share/nemo-python/extensions/nemo_terminal.py", line 247, in change_directory
    self.term.feed_child(eraselinekeys, len(eraselinekeys))
TypeError: Vte.Terminal.feed_child() takes exactly 2 arguments (3 given)
```

Inspecting line 247 in /usr/share/nemo-python/extensions/nemo_terminal.py:
```python
self.term.feed_child(eraselinekeys, len(eraselinekeys))
```
It seems the second parameter is not needed. After removing it, it still didn't work since there was a similar error on line 253. After removing the second parameter on that line it worked!

    sudo nano /usr/share/nemo-python/extensions/nemo_terminal.py
    CHANGE LINE 247 FROM
            self.term.feed_child(eraselinekeys, len(eraselinekeys))
    TO
            self.term.feed_child(eraselinekeys)
    END CHANGE
    CHANGE LINE 253 FROM
            self.term.feed_child(cdcmd, len(cdcmd))
    TO
            self.term.feed_child(cdcmd)
    END CHANGE

Note: I also went through and removed the second parameter in all other instances of feed_child (lines 101, 232, 234, 247, 253, 257, 408). The resulting nemo_terminal.py is attached: [nemo_terminal.py.txt](https://github.com/linuxmint/nemo-extensions/files/2096609/nemo_terminal.py.txt)

This problem looks like it is caused by a mismatch of the nemo-terminal code with the Vte.Terminal() that is being imported from gi.repository (although I don't know what these are). This solution seems like it should work on any distro.

**Comment by claudiux** [comment]
This doesn't work with Mint 19 beta and Cinnamon 3.8.4.

Did you do anything else?

**Comment by johanvandegriff** [comment]
I'm actually running Manjaro i3, not cinnamon, I just like the nemo file manager. To get the "open in terminal" feature working (I did that a while ago before enabling the nemo-terminal), I had to do:
```bash
gsettings set org.cinnamon.desktop.default-applications.terminal exec 'gnome-terminal'
```

It seems like the nemo terminal is independent of this setting since it creates its own terminal rather than calling an existing one. Your problem is probably different since it is a different version of Cinnamon/Nemo. Did you try killing the nemo process and running it in the terminal to be able to see the errors?

**Comment by claudiux** [comment]
I have the same parameter `org.cinnamon.desktop.default-applications.terminal` (which doesn't affect the nemo-terminal).
Yes, I killed the nemo process before running nemo with command `nemo`.
Errors appear only after removing of all parameters `len(...)`.

**Comment by claudiux** [comment]
Mint 19 with vte 2.91, all packages up to date.

While executing `sleep 10`: do_wait

While executing `read`: wait_woken

While executing `sudo apt-get install numlockx`, awaiting password entry: do_wait; while installing numlockx: do_wait.

Before and after these actions: poll_schedule_timeout.

**Comment by mtwebster** [comment]
Thanks - I have those covered, I meant any other people if they try using the patch I made yesterday, and still have problems.  I would imagine just kernel versions would make the most difference, I'll try to look into it further myself.

**Comment by leigh123linux** [comment]
It returns 0 here for everything

```
leigh    22127  3.3  0.3 1047620 64344 ?       Sl   07:26   0:00          |       \_ nemo
leigh    22137  0.0  0.0 220048  5168 pts/0    Ss+  07:26   0:00          |           \_ /bin/bash
```

rest

```
$ cat /proc/22137/wchan 
core_sys_select
```

sleep, read and installing all return 0

```
$ cat /proc/22137/wchan 
0
```

system info 

Fedora 29 (Rawhide)

```
$ uname -r
4.17.0-1.fc29.x86_64
```
```
$ rpm -qa vte*
vte291-0.52.2-1.fc29.x86_64
vte-profile-0.52.2-1.fc29.x86_64
```

**Comment by mtwebster** [comment]
Okay screw this, time to find a better way, maybe with a pipe

**Comment by leigh123linux** [comment]
@mtwebster Maybe looking at this will give you some new ideas.

https://github.com/flozz/nautilus-terminal

**Comment by leigh123linux** [comment]
80fbb75 doesn't fix the issue here

```
Traceback (most recent call last):
  File "/usr/share/nemo-python/extensions/nemo_terminal.py", line 444, in _on_crowbar_parent_set
    nterm.change_directory(self._uri)
  File "/usr/share/nemo-python/extensions/nemo_terminal.py", line 241, in change_directory
    self.term.feed_child(eraselinekeys, len(eraselinekeys))
TypeError: Vte.Terminal.feed_child() takes exactly 2 arguments (3 given)
```

**Comment by mtwebster** [comment]
3d834256650

**Comment by claudiux** [comment]
Yessss ! Thank you @mtwebster ! This is perfect.
